codec/packer.go

```diff
@@ -139,6 +140,49 @@ func (p *Packer) UnpackInt(required bool) int {
 	return int(v)
 }
 
+func (p *Packer) PackUvarInt(v uint64) {
+	buf := make([]byte, binary.MaxVarintLen64)
+	binary.PutUvarint(buf, v)
+	buf = ZeroOut(buf)
+	p.PackBytes(buf)
```

Comment:
PackBytes will wrap this buf in an integer length. That will be much larger in total than just using the uint64.

Reply:
There is good logic for how to properly do this inside of `merkledb`: https://github.com/ava-labs/avalanchego/blob/8d18b618be290b0e0fedbbedc76398afba68e6ec/x/merkledb/codec.go#L255-L272

Reply:
You may need to access the packer raw bytes to do this properly JFYI

Reply:
Updated and wrote a fuzz test to verify that encoding a number and decoding it back yields the same number, and that the bytes representation is never exceeding 10 bytes for any random 64 bit. I don't think that we need to add this check https://github.com/ava-labs/avalanchego/blob/8d18b618be290b0e0fedbbedc76398afba68e6ec/x/merkledb/codec.go#L261-L268 since it really looks like that empty bytes should be ignored here but I might be wrong https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/encoding/binary/varint.go;l=78-82